src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -99,12 +120,28 @@ public override TestResult Execute(TestExecutionContext context)
         {
             try
             {
-                if (!Task.Run(() => context.CurrentResult = innerCommand.Execute(context)).Wait(_timeout))
+                var testExecution = RunTestOnSeparateThread(context);
+
+                Wait(testExecution, _timeout);
+
+                if (testExecution.IsCompleted)
                 {
-                    context.CurrentResult.SetResult(new ResultState(
-                        TestStatus.Failed,
-                        $"Test exceeded Timeout value {_timeout}ms.",
-                        FailureSite.Test));
+                    context.CurrentResult = testExecution.GetAwaiter().GetResult();
+                }
+                else
+                {
+                    if (_debugger.IsAttached)
+                    {
+                        // Wait for the test to complete
+                        context.CurrentResult = testExecution.GetAwaiter().GetResult();
```

Comment:
Consider merging the duplicate blocks: `f (Task.WaitAny(new[] { testExecution }, _timeout) != -1 || _debugger.IsAttached)`